cdap-watchdog/src/main/java/io/cdap/cdap/logging/plugins/RollingLocationLogAppender.java

```diff
@@ -295,4 +315,46 @@ public void setFileMaxInactiveTimeMs(long fileMaxInactiveTimeMs) {
   public long getFileMaxInactiveTimeMs() {
     return fileMaxInactiveTimeMs;
   }
+
+  //Implement the class that handles delegation of locationOutputStream to outputStream defined in OutputStreamAppender
+  private class DelegatingOutputStream extends OutputStream {
+    private OutputStream delegatee;
+    DelegatingOutputStream(OutputStream outputStream) {
+      setDelegatee(outputStream);
+    }
+
+    public void setDelegatee(@Nullable OutputStream delegatee) {
+      this.delegatee = delegatee;
+    }
+
+    //Override the methods declared by the OutputStream class to invoke delegatee's methods
+    @Override
+    public void write(byte b[]) throws IOException {
+      this.delegatee.write(b);
```

Comment:
If you check for null everywhere, please check here as well

Reply:
Done